src/ui/map.js

```diff
@@ -2383,7 +2383,7 @@ class Map extends Camera {
      * @private
      */
     _requestRenderFrame(callback: () => void): TaskID {
-        this._update();
+        this.triggerRepaint();
```

Comment:
This has been around for 2 years, so it seems to me its the unlikely this broke it.

Can we change it so that in `handler_manager.js` we trigger triggerRepaint for `click`'s instead of always using _requestRenderFrame?

Reply:
this specifically didn't break the Studio, but the combination of this code and also handler_manager did.. The sourceDirty was set during the `mouseup` (which happens just before the click).. So the problem is not actually happening in case of click..

Reply:
Yes even though this code was there for a while there may very well have been a side effect from the gesture improvements as it was a fairly large change. By directly calling triggerRepaint instead of _update, we're effectively skipping those two statements every time we request a frame render:

```
this._styleDirty = this._styleDirty || updateStyle;
this._sourcesDirty = true;
```
This could have a similar side effect as it could trigger other types of events and it's a very high-level operation change. So the question would be whether [these code paths](https://github.com/mapbox/mapbox-gl-js/search?q=_requestRenderFrame&unscoped_q=_requestRenderFrame) strictly requires those two operations to be done, and whether there are any event triggered whenever any of those two booleans change.

Reply:
Another approach we can take is to actually figure out if a render frame and avoid calling for render frame. I think figuring this out might be tricky and might break some other flow.. It probably is safer to avoid marking the source as dirty. I tested `ease to` functionality which could be effected but didn't see any problem there.. Also ran the tests:
yarn test &&
yarn run test-expressions &&
yarn run test-query &&
yarn run test-render

@ansis

Reply:
Yes the one that was the most concerning would have been ease to (The other code path being in handler manager), thanks for testing that! I think we should be safe here.

Reply:
I think basic easeTo will continue to work, but there are some additional use cases like
- binding a `map.once('idle',` at right after triggering easeTo to detect when the animation ends, and map is ready with placed labels.
- easestart, easing and easeend events
- `map.loaded()` state check inside these various callbacks.

Reply:
Ok so it really seems like we should be going for adding this.map.triggerRepaint() + queue render task from handler_manager (basically inlining the code path that was changed here) to be free from introducing any other issue.